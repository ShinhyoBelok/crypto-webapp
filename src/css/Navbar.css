nav {
  align-items: center;
  color: white;
  background-color: #ec4c8b;
  justify-content: space-between;
  padding: 1rem;
}
nav h1 {
  font-weight: 300;
  font-size: medium;
}
nav .icons {
  gap: 0.5rem;
}
nav img {
  width: 1.2rem;
  filter: brightness(0) invert(1);
}
nav .arrow {
  align-items: center;
  gap: 0.5rem;
}
nav .arrow h2 {
  font-weight: 400;
  font-size: 1rem;
}

.backArrow {
  border: solid white;
  border-width: 0 3px 3px 0;
  display: inline-block;
  padding: 3px;
  transform: rotate(135deg);
  -webkit-transform: rotate(135deg);
}