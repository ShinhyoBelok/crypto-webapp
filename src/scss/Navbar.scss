nav {
  align-items: center;
  color: white;
  background-color: #ec4c8b;
  justify-content: space-between;
  padding: 1rem;
  h1 {
    font-weight: 300;
    font-size: medium;
  }
  .icons {
    gap: 0.5rem;
  }
  img {
    width: 1.2rem;
    filter: brightness(0) invert(1);
  }
  .arrow {
    align-items: center;
    gap: 0.5rem;
    h2 {
      font-weight: 400;
      font-size: 1rem;
    }
  }
}

.backArrow {
  border: solid white;
  border-width: 0 3px 3px 0;
  display: inline-block;
  padding: 3px;
  transform: rotate(135deg);
  -webkit-transform: rotate(135deg);
}