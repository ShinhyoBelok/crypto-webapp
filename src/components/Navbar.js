import React from 'react';
import { bool, string } from 'prop-types';
import { useNavigate } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { cryptoDetailFalse } from '../redux/crypto/crypto';
import '../css/Navbar.css';

export default function Navbar(props) {
  const {
    name,
    displayDetail,
  } = props;
  let year = '2023';
  if (displayDetail) {
    year = '';
  }
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const state = useSelector((state) => state.cryptos);
  const crypto = state.filter((crypto) => crypto.displayDetail === true);
  const backArrowClickHandler = () => {
    if (crypto.length) {
      const { id } = crypto[0];
      dispatch(cryptoDetailFalse({ id }));
      navigate('/');
    }
  };

  return (
    <nav className="flex">
      <div className="arrow flex">
        <div onClick={() => { backArrowClickHandler(); }} role="presentation" className="backArrow" />
        <h2>{year}</h2>
      </div>
      <h1>
        Crypto&ensp;
        {name}
      </h1>
      <div className="icons flex">
        <img src="https://cdn-icons-png.flaticon.com/512/25/25682.png" alt="mic" />
        <img src="https://cdn-icons-png.flaticon.com/512/3019/3019014.png" alt="settings" />
      </div>
    </nav>
  );
}

Navbar.propTypes = {
  name: string,
  displayDetail: bool,
};

Navbar.defaultProps = {
  name: 'Currency',
  displayDetail: false,
};
