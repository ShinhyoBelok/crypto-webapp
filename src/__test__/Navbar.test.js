import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Navbar from '../components/Navbar';

describe('Navbar component', () => {
  test('Test Navbar button render correctly', () => {
    const navbar = render(
      <Provider store={store}>
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      </Provider>,
    );
    expect(navbar).toMatchSnapshot();
  });
});
