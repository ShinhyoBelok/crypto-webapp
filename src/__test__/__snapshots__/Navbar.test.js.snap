// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component Test Navbar button render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="flex"
      >
        <div
          class="arrow flex"
        >
          <div
            class="backArrow"
            role="presentation"
          />
          <h2>
            2023
          </h2>
        </div>
        <h1>
          Crypto 
          Currency
        </h1>
        <div
          class="icons flex"
        >
          <img
            alt="mic"
            src="https://cdn-icons-png.flaticon.com/512/25/25682.png"
          />
          <img
            alt="settings"
            src="https://cdn-icons-png.flaticon.com/512/3019/3019014.png"
          />
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="flex"
    >
      <div
        class="arrow flex"
      >
        <div
          class="backArrow"
          role="presentation"
        />
        <h2>
          2023
        </h2>
      </div>
      <h1>
        Crypto 
        Currency
      </h1>
      <div
        class="icons flex"
      >
        <img
          alt="mic"
          src="https://cdn-icons-png.flaticon.com/512/25/25682.png"
        />
        <img
          alt="settings"
          src="https://cdn-icons-png.flaticon.com/512/3019/3019014.png"
        />
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
